presto-main/src/test/java/com/facebook/presto/operator/scalar/TestDateTimeFunctions.java

```diff
@@ -190,6 +191,25 @@ public void testFromUnixTimeWithOffset()
     }
 
     @Test
+    public void testFromUnixTimeWithTimeZone()
+    {
+        DateTime dateTime = new DateTime(2001, 1, 22, 3, 4, 5, 0, DATE_TIME_ZONE);
+        double seconds = dateTime.getMillis() / 1000.0;
+
+        String zoneId = "Asia/Tokyo";
+        DateTime expected = new DateTime(dateTime, getDateTimeZone(getTimeZoneKey(zoneId)));
+        assertFunction(format("from_unixtime(%s, '%s')", seconds, zoneId), TIMESTAMP_WITH_TIME_ZONE, toTimestampWithTimeZone(expected));
```

Comment:
Make sure your test covers both positive and negative zones. Make sure your test covers both named time zone and offset time zone.

Additionally, I would use something plainer in tests. (e.g. `assertFunction("from_unixtime(7200, 'Asia/Shanghai')", new DateTime(1970, 1, 1, 10, 0, 0, DateTimeZone.forID("Asia/Shanghai")))"`) Using a unix time near 0 should make the tests fairly readable.